File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> mp;
        for (int i = 0;i<nums.size();i++){
            mp[nums[i]]++;
        }

        // priority_queue<pair<int,int>> maxHeap;
        priority_queue<pair<int,int>, vector<pair<int,int>>, greater<pair<int,int>>> minHeap;

        for (auto it: mp){
            minHeap.push({it.second, it.first});

            if(minHeap.size() > k){
                minHeap.pop();
            }

        }
        vector<int> res;
        while (!minHeap.empty()){
            res.push_back(minHeap.top().second);
            minHeap.pop();

        }
        return res;

 



        
    }
};

File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        dic = dict()

        for i in nums:
            if i in dic:
                dic[i] +=1 
            else:
                dic[i] = 1 
        ans = []
        sorted_dic = {k:v for k,v in sorted(dic.items(), key=lambda x:x[1], reverse=True)}
        print(sorted_dic)
        for i in sorted_dic: 
            ans.append(i)
            k-=1 
            if k==0:
                return ans 
                

        

